src/azurefluent/TransformerJvaf.cs

```diff
@@ -153,9 +152,9 @@ private void AppendInnerToTopLevelType(IModelType type, CodeModel serviceClient)
             if (compositeType != null && !compositeType.IsResource)
             {
                 compositeType.IsInnerModel = true;
-                foreach (var t in serviceClient.ModelTypes)
+                foreach (var t in serviceClient.ModelTypes.Where(mt => !(mt as CompositeTypeJvaf).IsInnerModel))

```

Comment:
If the cast is expected to always succeed it's good to use the cast syntax instead of `as`